docs/library/layout/match.md

```diff
@@ -0,0 +1,266 @@
+provided in this scenario.
+
+
+## Handling Multiple Conditions
+
+`rx.match` excels in efficiently managing multiple conditions and their corresponding components, 
+providing a cleaner and more readable alternative compared to nested `rx.cond` structures.
+
+Consider the following example:
+
+```python demo exec
+from typing import List
+
+options: List[str] = ["red", "green", "blue", "yellow", "orange", "black", "white"]
+
+class MultiMatchState(rx.State):
+    color: str = ""
+
+
+
+def multi_match_demo():
+    return rx.vstack(
+        rx.match(
+            MultiMatchState.color,
+            ("red", "yellow", "blue", rx.text("primary color")),
+            ("green","orange", rx.text("secondary color")),
+            rx.text("neither primary nor secondary.")
+        ),
+        rx.select(
+            options,
+            placeholder="Select a color.",
+            on_change=MultiMatchState.set_color,
+        ),
+    )
+
+
+```
+
+In a match case tuple, you can specify multiple conditions. The last value of the match case 
+tuple is automatically considered as the return value. It's important to note that a match case
+tuple should contain a minimum of two elements: a match case and a return value. 
+The following code snippet will result in an error:
+
+```python
+rx.match(
+            MatchState.color,
+            ("red",),
+            ("green", rx.text("green color selected")),
+        )
+
+```
+## Usage as Props
+Similar to rx.cond, rx.match can be used as prop values, allowing dynamic behavior for UI components:
+
+```python demo exec
+class MatchPropState(rx.State):
+    value: int = 0
+    
+    def incr(self):
+        self.value += 1
+    
+    def decr(self):
+        self.value -= 1
+
+
+def match_prop_demo_():
+    return rx.vstack(
+        rx.text(
+            MatchPropState.value, 
+            color= rx.match(
+                MatchPropState.value,
+                (1, "red"),
+                (5, "blue"),
+                (10, "green"),
+                (15, "orange"),
+                "black"
+                
+            )
+        ),
+        rx.hstack(
+            rx.button("increment", on_click=MatchPropState.incr),
+            rx.button("decrement", on_click=MatchPropState.decr),
+        ),
+    )
+
+
+```
+
+In the example above, the color property of the text component containing `State.value` changes to red when 
+`state.value` is 1, blue when its 5, green when its 5, orange when its 15 and black for any other value.
+
+The example below also shows handling multiple conditions with the match component as props.
+
+```python demo exec
+class MatchMultiPropState(rx.State):
+    value: int = 0
+    
+    def incr(self):
+        self.value += 1
+    
+    def decr(self):
+        self.value -= 1
+
+
+def match_multi_prop_demo_():
+    return rx.vstack(
+        rx.text(
+            MatchMultiPropState.value, 
+            color= rx.match(
```

Comment:
A little hard to tell the color is changing maybe make it more apparent somehow, possible change a background color instead